### DISCUSSION TITLE
PostGIS plugin

### DISCUSSION THREAD

**Discussion by khangphung-vl** [mikro-orm/mikro-orm#6274] [General]
Hello @B4nan,
I'm a fan of Mikro, simply the best ORM I've ever used. I've come through this [reddit post ](https://www.reddit.com/r/node/comments/14u9tnd/to_mikroormpostgres_crowd_did_anyone_write_at/)when trying to find whether the PostGIS can be use with Mikro.

I'm interested in implementing and want to ask for your opinion on what does the term **"plugin"** means, can you describe it a bit. I have little experience in package/tool/... development, but I want to give it a try.

**Comment by B4nan** [comment]
For postgis I would imagine a set of custom type implementations for the geometry types, I don't think we need a plugin per se, since there is nothing to do on ORM init, people need to use those types where appropriate. 

One example of such type is the PointType in the docs:

https://mikro-orm.io/docs/custom-types#advanced-example---pointtype-and-wkt

I've never used those types in a production app myself, so can't give you any details, you'll have to research things yourself. I can imagine some of those things will be platform specific, but maybe not. If so, you could consider having some abstraction in those types so people can use one type for all databases, or going for a simple type per driver approach.

**Reply by khangphung-vl** [reply]
Thanks.

**Reply by darkbasic** [reply]
I have it implemented in one of my apps and I was planning on releasing a package to plug and play it but never had time to do so. Maybe during the holidays...
